server/src/controllers/seasonController.ts

```diff
@@ -353,38 +354,60 @@ seasonController.put('/:id', async (req: Request, res: Response) => {
  */
 seasonController.delete('/:id', async (req: Request, res: Response) => {
   try {
-    const id = req.params.id;
-    const seasonExists = await prisma.seasons.findUnique({
+    const id = +req.params.id;
+    const season = await prisma.seasons.findUnique({
       where: {
-        seasonid: Number(id),
+        seasonid: id,
+        deletedat: null,
+      },
+      include: {
+        seasonsubscriptiontypes: {
+          include: {
+            subscriptions: true,
+          },
+        },
       },
     });
-    if (!seasonExists) {
-      res.status(404).json({error: 'season not found'});
 
-      return;
+    if (!season) {
+      return res.status(400).json({error: 'season not found'});
```

Comment:
If this is changed to a 400, would you mind updating the swagger doc comment to match.

Reply:
Fixed